src/GlobalAccount.ts

```diff
@@ -35,6 +37,145 @@ export function setGlobalNetworkAccount(account: AccountDB.AccountCopy): void {
+    activeConfigChanges.add(changeHash)
+    const changeObj = change.change
+    const appData = change.appData
+
+    // If there is initShutdown change, if the latest cycle is greater than the cycle of the change, then skip it
+    if (changeObj['p2p'] && changeObj['p2p']['initShutdown'] && change.cycle !== cycleNumber) continue
+
+    const newChanges = pruneNetworkChangeQueue(changes, cycleNumber)
+    networkAccount.data.listOfChanges = newChanges
+    // https://github.com/shardeum/shardeum/blob/c449ecd21391747c5b7173da3a74415da2acb0be/src/index.ts#L6958
+    // Increase the timestamp by 1 second
+    // networkAccount.data.timestamp += 1000
+
+    if (appData) {
+      updateNetworkChangeQueue(networkAccount.data, appData)
+      console.dir(networkAccount.data, { depth: null })
+      // https://github.com/shardeum/shardeum/blob/c449ecd21391747c5b7173da3a74415da2acb0be/src/index.ts#L6889
+      // Increase the timestamp by 1 second
+      // networkAccount.data.timestamp += 1000
+    }
+
+    networkAccount.hash = accountSpecificHash(networkAccount.data)
+    networkAccount.timestamp = networkAccount.data.timestamp
+    Logger.mainLogger.debug('updateGlobalNetworkAccount', networkAccount)
+    await AccountDB.updateAccount(networkAccount)
+    setGlobalNetworkAccount(networkAccount)
+  }
+  if (activeConfigChanges.size > 0) {
+    // clear the entries from appliedConfigChanges that are no longer in the changes list
+    for (const changeHash of appliedConfigChanges) {
+      if (!activeConfigChanges.has(changeHash)) {
+        appliedConfigChanges.delete(changeHash)
+      }
+    }
+  }
+}
+
+const generatePathKeys = (obj: any, prefix = ''): string[] => {
+  /* eslint-disable security/detect-object-injection */
+  let paths: string[] = []
+
+  // Loop over each key in the object
+  for (const key of Object.keys(obj)) {
+    // If the value corresponding to this key is an object (and not an array or null),
+    // then recurse into it.
+    if (obj[key] !== null && typeof obj[key] === 'object' && !Array.isArray(obj[key])) {
+      paths = paths.concat(generatePathKeys(obj[key], prefix + key + '.'))
+    } else {
+      // Otherwise, just append this key to the path.
+      paths.push(prefix + key)
+    }
+  }
+  return paths
+  /* eslint-enable security/detect-object-injection */
+}
+
+const pruneNetworkChangeQueue = (
+  changes: NetworkConfigChanges[],
+  currentCycle: number
+): NetworkConfigChanges[] => {
+  const configsMap = new Map()
+  const keepAliveCount = config.configChangeMaxChangesToKeep
+  for (let i = changes.length - 1; i >= 0; i--) {
+    const thisChange = changes[i]
+    let keepAlive = false
+
+    let appConfigs = []
+    if (thisChange.appData) {
+      appConfigs = generatePathKeys(thisChange.appData, 'appdata.')
+    }
+    const shardusConfigs: string[] = generatePathKeys(thisChange.change)
+
+    const allConfigs = appConfigs.concat(shardusConfigs)
+
+    for (const config of allConfigs) {
```

Comment:
I think this was cut an pasted from the validator o that the archive server can understand these values.   
the goal of prune work changes is to keep the last value when there have been past changes.
i.e  what if we change some config daily reward rates.  this will prune out the old change history for that config

Reply:
I will ask Red to help double check where the code was transplanted